cmd/webhook/main.go

```diff
@@ -32,9 +34,43 @@ func main() {
 	// Avoid "logging before flag.Parse" errors from glog
 	flag.CommandLine.Parse([]string{})
 
+	// parse the command line flags to pull out the tls-cert-file
+	// argument. This flag will be parsed by code inside cmd.RunAdmissionServer
+	// so no need to pass it through the call stack or have nice errors
+	tlsflagSet := flag.NewFlagSet("tls", flag.ContinueOnError)
+	tlsflagVal := tlsflagSet.String("tls-cert-file", "", "")
+	tlsflagSet.Parse(os.Args[1:])
+	if *tlsflagVal != "" {
+		runfilewatch(*tlsflagVal)
+	}
+
 	cmd.RunAdmissionServer(
 		certHook,
 		issuerHook,
 		clusterIssuerHook,
 	)
 }
+
+func runfilewatch(filename string) {
+	info, err := os.Stat(filename)
+	if err != nil {
+		// missing TLS cert file will get turned into a proper error later
+		return
+	}
+	modtime := info.ModTime()
+	go func() {
+		for {
+			time.Sleep(1 * time.Minute)
+			info, err := os.Stat(filename)
+			if err != nil {
+				continue
+			}
+			if info.ModTime().After(modtime) {
+				// let the k8s scheduler restart us
+				// TODO(dmo): figure out if there's a way to do this with clean
+				// shutdown
+				os.Exit(0)
```

Comment:
Can we log a message to explain why we're exiting? In the event of a bug here, this will make it a lot clearer to users what's up 😄

Reply:
done